thirdparty/mcap/mcap/reader.inl

```diff
@@ -360,6 +361,13 @@ void McapReader::reset_() {
   parsedSummary_ = false;
 }
 
+void McapReader::clear_attachments_() {
+    for (auto& [attachment_name, attachment] : attachments_) {
+        std::free((void*)attachment.data);
+    }
+    attachments_.clear();
+}
+
```

Comment:
The function `.clear()` calls the destructor of each element inside the map. Wouldn't it make more sense to override the destructor of the Attachment object to free `attachment.data`? In that way you would avoid having a `clear_attachments_` function and simply call `attachments_.clear()` like with every other object.

Reply:
I don't think that would be valid, because elsewhere (every place other than when destroying our added map) it would be calling a free that is already called somewhere else (there is no way of checking if already freed easily, perhaps adding an attribute, but too involved and risky).

Another option would be to inherit from Attachment and add that destructor only there (and have a map of children). I would not go for that, our changes to the mcap library should be minimal.